connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await target_file.read()).strip()
+        try:
+            await remove(temp_filename)
+        except Exception as exception:
+            self._logger.warning(
+                f"Could not remove file: {temp_filename}. Error: {exception}"
+            )
+
+        return document
+
+    async def get_content(self, doc, timestamp=None, doit=False):
+        attachment_size = doc["file_size"]
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = doc["file_name"]
+
+        attachment_extension = (
+            attachment_name[attachment_name.rfind(".") :]  # noqa
+            if "." in attachment_name
+            else ""
+        )
+
+        if not self._pre_checks_for_get_content(
+            attachment_extension=attachment_extension,
+            attachment_name=attachment_name,
+            attachment_size=attachment_size,
+        ):
+            return
+
+        self._logger.debug(f"Downloading {attachment_name}")
+        return await self._get_document_with_content(doc)
+
+    async def fetch_previous_meeting_details(self, meeting_id):
+        previous_meeting = await self.client.get_past_meeting(meeting_id=meeting_id)
+
+        if not previous_meeting:
+            return
+
+        participants = []
+        async for participant in self.client.get_past_meeting_participants(
+            meeting_id=meeting_id
+        ):
+            participants.append(participant)
+
+        previous_meeting["participants"] = participants
+        return previous_meeting
+
+    async def fetch_recordings(self, user_id):
+        async for recording in self.client.get_recordings(user_id=user_id):
+            yield recording
+
+        async for recording in self.client.get_trash_recordings(user_id=user_id):
+            yield recording
+
+    async def get_docs(self, filtering=None):
+        async for user in self.client.get_users():
+            yield self._format_doc(doc=user, doc_time=iso_utc()), None
+
+            async for live_meeting in self.client.get_meetings(
+                user_id=user.get("id"), meeting_type="live"
+            ):
+                yield self._format_doc(
+                    doc=live_meeting, doc_time=live_meeting.get("created_at")
+                ), None
+
+            async for upcoming_meeting in self.client.get_meetings(
+                user_id=user.get("id"), meeting_type="upcoming_meetings"
+            ):
+                yield self._format_doc(
+                    doc=upcoming_meeting,
+                    doc_time=upcoming_meeting.get("created_at"),
+                ), None
+
+            async for previous_meeting in self.client.get_meetings(
+                user_id=user.get("id"), meeting_type="previous_meetings"
+            ):
+                if self.configuration["fetch_past_meeting_details"]:
```

Comment:
Currently `live_meeting`, `upcoming_meeting` & `previous_meeting` are indexed for all fetched_user and regarding `fetch_past_meeting_details` as per [slack discussion](https://crestdatasystems.slack.com/archives/C02AV5C8ZDE/p1692191817237099?thread_ts=1692005081.034619&cid=C02AV5C8ZDE) it will be fetching other information like [Details](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/pastMeetingDetails) & [Participants](https://developers.zoom.us/docs/api/rest/reference/zoom-api/methods/#operation/pastMeetingParticipants) for past meetings as more API calls needed